oneview_redfish_toolkit/api/volume.py

```diff
@@ -0,0 +1,202 @@
+# under the License.
+
+import collections
+from flask import abort
+from flask import g
+
+from flask_api import status
+from hpOneView.resources.resource import ResourceClient
+from oneview_redfish_toolkit.api.computer_system import ComputerSystem
+
+
+from oneview_redfish_toolkit.api.redfish_json_validator \
+    import RedfishJsonValidator
+
+import oneview_redfish_toolkit.api.status_mapping as status_mapping
+
+
+class Volume(RedfishJsonValidator):
+    """Creates a Volume Redfish dict
+
+        Populates self.redfish with Volume data
+    """
+
+    SCHEMA_NAME = 'Volume'
+    METADATA_INFO = "/redfish/v1/$metadata#Volume.Volume"
+
+    def __init__(self, data):
+        """Volume constructor
+
+            Populates self.redfish and validates the result
+
+            Args:
+                data: a dict with Redfish's Volume data
+        """
+        super().__init__(self.SCHEMA_NAME)
+
+        self.redfish.update(data)
+
+        self.redfish["@odata.type"] = self.get_odata_type()
+        self.redfish["@odata.context"] = self.__class__.METADATA_INFO
+
+        self._validate()
+
+    @staticmethod
+    def build_volume_details(uuid, volume_id):
+        """Returns a Volume with the contents of data from an Oneview
+
+            Args:
+                uuid: Server Profile uuid
+                volume_id: Volume id
+
+        """
+
+        server_profile = g.oneview_client.server_profiles.get(uuid)
+        sas_logical_jbod = get_sas_logical_jbod_by_volumeid(server_profile,
+                                                            volume_id)
+
+        if sas_logical_jbod is None:
+            abort(status.HTTP_404_NOT_FOUND, "Volume {} not found"
+                  .format(volume_id))
+
+        sas_Logical_Interconnect_Uri = \
+            sas_logical_jbod["sasLogicalInterconnectUri"]
+
+        drivepaths = []
+        drivepath = None
+        for logical_Drive_Bay_Uri in sas_logical_jbod["logicalDriveBayUris"]:
+            drivepath = get_drive_path_from_logical_Drive_Bay_Uri(
+                logical_Drive_Bay_Uri)
+            drivepaths.append(drivepath)
+
+        drive_enclosure_uri = \
+            get_drive_enclosure_uri_from_sas_Logical_Interconnect(
+                sas_Logical_Interconnect_Uri)
+
+        drive_enclosure_object = get_drive_enclosure_object(
+            drive_enclosure_uri)
+
+        drivebayuris = []
+
+        drivebayuri = None
+        drivepath = None
+        flag = False
+        for path in drivepaths:
+            flag = False
+            for drivebay in drive_enclosure_object["driveBays"]:
+                for drivepath in drivebay["drive"]["drivePaths"]:
+                    if drivepath == path:
+                        flag = True
+                        drivebayuri = drivebay["uri"]
+                        drivebayuris.append(drivebayuri)
+                        break
+                if flag:
+                    break
```

Comment:
You can do this code block using just two **for** loops instead of three, by removing the more external **for** loop, and on the **if** at line 102, you could change it to "**if drivepath in drivepaths:**"
And you can extract this code block into a new method for remove the flags.